src/clj/fluree/db/query/api.cljc

```diff
@@ -35,7 +35,8 @@
   [query {:keys [identity did issuer] :as override-opts}]
   (update query :opts (fn [{:keys [max-fuel meta] :as opts}]
                         ;; ensure :max-fuel key is present
-                        (-> (assoc opts :max-fuel max-fuel)
+                        (-> (merge opts override-opts)
+                            (assoc :max-fuel max-fuel)
```

Comment:
Oh, I just realized that the `(assoc :max-fuel max-fuel)` is using the `:max-fuel` from `opts`, so you'll be clobbering the `override-opts`. I think we can simplify the whole thread form:

```
(-> (assoc opts :max-fuel max-fuel)
      (merge opts override-opts)
      ;; get rid of :did, :issuer opts
      (update :identity #(or % (:did opts) (:issuer opts)))
      (dissoc :did :issuer))

Reply:
Ah, that makes sense, good catch!